File: exercicios/respostas/lista01/exercicio_1.py
#### Código Base (Python):
import time
import random

def encontrar_maximo(lista):
    """
    Encontra o maior elemento em uma lista
    
    Args:
        lista: Lista de inteiros
        
    Returns:
        int: Maior elemento da lista
    """
    maximo = 0
    for inteiro in lista:
        if maximo < inteiro:
            maximo = inteiro
    return maximo

def medir_tempo(lista):
    """Mede o tempo de execução do algoritmo"""
    inicio = time.time()
    maximo = encontrar_maximo(lista)
    fim = time.time()
    return fim - inicio

def main():
    # Testar com diferentes tamanhos
    tamanhos = [10, 100, 1000, 10000]
    
    for n in tamanhos:
        # Criar lista com valores aleatórios
        lista = [random.randint(0, 999) for _ in range(n)]
        
        tempo = medir_tempo(lista)
        print(f"Tamanho: {n}, Tempo: {tempo:.6f} segundos")

if __name__ == "__main__":
    main()

File: exercicios/respostas/lista01/Exercicio1.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

// Implementar esta função
int encontrar_maximo(int arr[], int n) {
    int maximo = 0;

    for (int i = 0; i < sizeof(arr); i++){
        if (maximo<=arr[i]){
            maximo = arr[i];
        }
    }
    return maximo;
}

// Função para medir tempo de execução
double medir_tempo(int arr[], int n) {
    clock_t inicio = clock();
    int max = encontrar_maximo(arr, n);
    clock_t fim = clock();
    return ((double)(fim - inicio)) / CLOCKS_PER_SEC;
}

int main() {
    // Testar com diferentes tamanhos
    int tamanhos[] = {10, 100, 1000, 10000};
    int num_testes = 4;
    
    for (int i = 0; i < num_testes; i++) {
        int n = tamanhos[i];
        int* arr = malloc(n * sizeof(int));
        
        // Preencher array com valores aleatórios
        for (int j = 0; j < n; j++) {
            arr[j] = rand() % 1000;
        }
        
        double tempo = medir_tempo(arr, n);
        printf("Tamanho: %d, Tempo: %f segundos\n", n, tempo);
        
        free(arr);
    }
    
    return 0;
}